ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/fillers/MaxLoopFlowFiller.java

```diff
@@ -96,42 +81,35 @@ private void buildLoopFlowConstraintsAndUpdateObjectiveFunction(RaoData raoData,
 
             maxLoopFlowLimit = Math.max(0.0, maxLoopFlowLimit - loopFlowConstraintAdjustmentCoefficient);
 
-            double commercialFlow;
-            //get commercial flow
-            if (!isLoopFlowApproximation) {
-                commercialFlow = loopFlowResult.getCommercialFlow(cnec);
-            } else {
-                commercialFlow = cnec.getExtension(CnecLoopFlowExtension.class).getLoopflowShift();
-            }
-
             //get MPVariable: flowVariable
             MPVariable flowVariable = linearProblem.getFlowVariable(cnec);
             if (Objects.isNull(flowVariable)) {
                 throw new FaraoException(String.format("Flow variable on %s has not been defined yet.", cnec.getId()));
             }
 
             MPVariable loopflowViolationVariable = linearProblem.addLoopflowViolationVariable(
-                0,
-                linearProblem.infinity(),
-                cnec
+                    0,
+                    linearProblem.infinity(),
+                    cnec
             );
 
-            // - MaxLoopFlow + LoopFlowShift <= flowVariable + loopflowViolationVariable <= POSITIVE_INF
+            double commercialFlow = cnec.getExtension(CnecResultExtension.class).getVariant(raoData.getWorkingVariantId()).getCommercialFlowInMW();
+            // - MaxLoopFlow + commercialFlow <= flowVariable + loopflowViolationVariable <= POSITIVE_INF
```

Comment:
For the iterations >=2, the commercialFlow is set during the systematic analaysis of iteration n-1 : OK

For iteration 1, I'm not sure that the commercialFlow has been properly set. I think that we need an 'initial loop-flow computation' at the beginning of the iterationLinearRao, no ?

Reply:
At the beginning of the RAO, it is done in the InitialSensitivityAnalysis.
For each new Leaf, it is done in Leaf.evaluate() if we should update the PSTs with topo changes

Reply:
Indeed, perfect then !